dashboard/client/src/pages/serve/ServeApplicationsListPage.tsx

```diff
@@ -73,132 +79,121 @@ export const ServeApplicationsListPage = () => {
-              </TableRow>
-            </TableHead>
-            <TableBody>
-              {filteredServeApplications
-                .slice(
-                  (page.pageNo - 1) * page.pageSize,
-                  page.pageNo * page.pageSize,
-                )
-                .map((application) => (
-                  <ServeApplicationRow
-                    key={application.name}
-                    application={application}
-                  />
-                ))}
-            </TableBody>
-          </Table>
-        </TableContainer>
-      </CollapsibleSection>
+          <CollapsibleSection
+            title="Serve applications"
+            startExpanded
+            className={classes.applicationsSection}
+          >
+            <TableContainer>
+              <div style={{ flex: 1, display: "flex", alignItems: "center" }}>
+                <Autocomplete
+                  style={{ margin: 8, width: 120 }}
+                  options={Array.from(
+                    new Set(
+                      allServeApplications.map((e) => (e.name ? e.name : "-")),
+                    ),
+                  )}
+                  onInputChange={(_: any, value: string) => {
+                    changeFilter(
+                      "name",
+                      value.trim() !== "-" ? value.trim() : "",
+                    );
+                  }}
+                  renderInput={(params: TextFieldProps) => (
+                    <TextField {...params} label="Name" />
+                  )}
+                />
+                <Autocomplete
+                  style={{ margin: 8, width: 120 }}
+                  options={Array.from(
+                    new Set(allServeApplications.map((e) => e.status)),
+                  )}
+                  onInputChange={(_: any, value: string) => {
+                    changeFilter("status", value.trim());
+                  }}
+                  renderInput={(params: TextFieldProps) => (
+                    <TextField {...params} label="Status" />
+                  )}
+                />
+                <TextField
+                  style={{ margin: 8, width: 120 }}
+                  label="Page Size"
+                  size="small"
+                  defaultValue={10}
+                  InputProps={{
+                    onChange: ({ target: { value } }) => {
+                      setPage("pageSize", Math.min(Number(value), 500) || 10);
+                    },
+                    endAdornment: (
+                      <InputAdornment position="end">Per Page</InputAdornment>
+                    ),
+                  }}
+                />
+              </div>
+              <div style={{ display: "flex", alignItems: "center" }}>
+                <Pagination
+                  count={Math.ceil(
+                    filteredServeApplications.length / page.pageSize,
+                  )}
+                  page={page.pageNo}
+                  onChange={(e, pageNo) => setPage("pageNo", pageNo)}
```

Comment:
nit: we should probably change this state hook eventually to be typed instead of an object